wagtail/admin/tests/test_pages_views.py

```diff
@@ -2602,19 +2609,31 @@ def test_page_move_bad_permissions(self):
         self.user.save()
 
         # Get move page
-        response = self.client.get(reverse('wagtailadmin_pages:move', args=(self.test_page.id, )))
+        response = self.client.get(reverse('wagtailadmin_pages:move', args=(self.test_page_a.id, )))
 
         # Check that the user received a 403 response
         self.assertEqual(response.status_code, 403)
 
     def test_page_move_confirm(self):
         response = self.client.get(
-            reverse('wagtailadmin_pages:move_confirm', args=(self.test_page.id, self.section_b.id))
+            reverse('wagtailadmin_pages:move_confirm', args=(self.test_page_a.id, self.section_b.id))
         )
         self.assertEqual(response.status_code, 200)
 
+        response = self.client.get(
+            reverse('wagtailadmin_pages:move_confirm', args=(self.test_page_b.id, self.section_a.id))
+        )
+        # Duplicate slugs triggers a redirect with an error message.
+        self.assertEqual(response.status_code, 302)
+
+        response = self.client.get(reverse('wagtailadmin_home'))
+        for message in response.context['messages']:
```

Comment:
This doesn't correctly test for the presence of the message - it would also pass if `response.context['messages']:` was empty.